### DISCUSSION TITLE
Using downloadArchive API function to download an album as ZIP

### DISCUSSION THREAD

**Discussion by rhvarrier** [immich-app/immich#14227] [Q&A] [answered]
Hi all

I'm trying to write a Python script that downloads an album as a ZIP folder from the server given its album name.

Using getAllAlbumsgetAllAlbums API function I'm able to get the UUID for my album, then I tried using this uuid ("id" key in json output) with the function downloadArchive.

However when making the request I get the following output:

```
{"message":"Not found or no [asset.download](http://asset.download) access","error":"Bad Request","statusCode":400,"correlationId":"qggq896j"} 

```  
  
I make the request using the following snippet:
```
    url = _SERVER_URL+"/api/download/archive"
    payload = json.dumps({
        "assetIds": [
            f"{id}"
        ]
        })
    headers = {
    "Content-Type": "application/json",
    "x-api-key": f"{_API_KEY}",
    "Accept": "application/octet-stream"
    }
    response = requests.request("POST", url, headers=headers, data=payload)
    print(response.text)

```

And I get UUID for the first album with the following snippet:
```
    url = _SERVER_URL+"/api/albums"
    payload = {}
    headers = {
    "Accept": "application/json",
    "x-api-key": f"{_API_KEY}"
    }
    response = requests.request("GET", url, headers=headers, data=payload)
    album_uuid = son.loads(response.text)[0]["id"]
```

Is this the correct approach ? Sorry this seems trivial, but this is the first I'm using a web API

**Comment by bo0tzz** [comment]
If I understand right, you're trying to use the album's ID in a spot where the IDs of assets should go.

**Reply by rhvarrier** [reply]
Thanks for the reply.
What I'm trying to download an album. I understand there is a function called downloadArchive that downloads something given it's uuid. So I was hopping that I can give this function the album Id and it will download the whole album. So is there a way to get the ID of the assets of a given album? I have found that the "assets" objet of getAllAlbums response is empty even if my albums is not

**Reply by bo0tzz** [reply, marked as answer]
The key in the download request body is pretty clearly called `assetIds` ;) The response from getAllAlbums deliberately doesn't include all the assets. You can get those with https://immich.app/docs/api/get-album-info.
